wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/ModalDialog.js

```diff
@@ -0,0 +1,424 @@
+							+ " Clicking outside the modal or pressing ESC will have no effect");
+				}
+			}
+		}
+	};
+
+	/** Detects clicks on secondary close buttons (SENODARY_CLOSE_SELECTOR) and forwards the click to the primary close button */
+	var secondaryCloseButtonBehavior = {
+		prepare: function(overlay, element, data) {
+			overlay.on("click", SECONDARY_CLOSE_SELECTOR, function(event) {
+				event.preventDefault();
+				event.stopPropagation();
+				findAndClickCloseButton(element);
+			});
+		}
+	}
+
+	/** Traps focus inside the modal window. */
+	var trapFocusInsideModalBehavior = {
+		prepare : function(overlay, element, data) {
+			overlay.on("keydown", function(e) {
+				if (e.which === 9) { // tab
+					var container = $(e.target).closest(CONTENT_SELECTOR);
+					var focusables = findTabbable(container);
+					var firstFocusable = focusables.get(0);
+					var lastFocusable = focusables.get(focusables.length - 1);
+
+					if (!e.shiftKey && e.target === lastFocusable) {
+						e.preventDefault();
+						firstFocusable.focus();
+					}
+					if (e.shiftKey && e.target === firstFocusable) {
+						e.preventDefault();
+						lastFocusable.focus();
+					}
+				}
+			});
+
+			overlay.on("DOMNodeRemoved.modal-dialog", function(e) {
+				// handles focus transitions when nodes are removed, for example
+				// a node that has focus is removed via an ajax update
+				window.setTimeout(function() {
+					// needs to run in timeout because the event will get called
+					// with the node that is being removed as active
+					var active = $(document.activeElement);
+					if (active.closest(CONTENT_SELECTOR).length === 0) {
+						// focus has been moved to something outside the modal,
+						// refocus
+						focusDefaultFocusable(element);
+					}
+				}, 0);
+			});
+		},
+		close : function(overlay, element, data) {
+			overlay.off("DOMNodeRemoved.modal-dialog");
+		}
+
+	};
+
+	var focusDefaultOnOpeningBehavior = {
+		open : function(overlay, element, data) {
+			focusDefaultFocusable(element);
+		}
+	};
+
+	var sizingBehavior = {
+		prepare : function(overlay, element, data) {
+			if (data.options.maxWidth) {
+				overlay.find(CONTAINER_SELECTOR).css({
+					maxWidth : data.options.maxWidth
+				});
+			}
+			if(data.options.maxHeight) {
+				overlay.find(SCROLL_SELECTOR).css({
+					maxHeight: data.options.maxHeight
+				});
+			}
+		}
+	};
+
+	var defaultOptions = {
+		validate : false,
+		console : window.console,
+		maxWidth : null,
+		maxHeight: null, //"80vh"
+		closeOnClickOutside: false,
+		closeOnEscape: true
+	};
+
+	var behaviors = [ appendBodyClassBehavior, returnFocusOnCloseBehavior, closeOnOverlayClickOrEscapeBehavior, addAriaAttributesBehavior,
+			trapFocusInsideModalBehavior, focusDefaultOnOpeningBehavior, sizingBehavior, secondaryCloseButtonBehavior ];
+
+	//
+	// Entry Methods
+	//
+
+	window.wicket = window.wicket || {};
```

Comment:
Why Wicket ns doesn't start from capital W here?

Reply:
no particular reason.

Reply:
We should change it to capital `W`. The idea here is to have one global entry in the JS DOM - `window.Wicket`